rdeprojector/controllers/rdeprojector_controller.go

```diff
@@ -87,7 +107,75 @@ func (r *RDEProjectorReconciler) reconcileNormal(ctx context.Context, rdeProject
 	capiYaml := entity["spec"].(map[string]interface{})["capiYaml"]
 	log.Info("Retrieved Capi Yaml", "capiYaml", capiYaml)
 
-	//TODO Apply CAPI YAML
+	// Retrieve kubeconfig of the cluster
+	kubeconfig := ctrl.GetConfigOrDie()
+
+	// Prepare a RESTMapper to find GVR
+	dc, err := discovery.NewDiscoveryClientForConfig(kubeconfig)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(dc))
+
+	// Get the dynamic client
+	dyn, err := dynamic.NewForConfig(kubeconfig)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	var decUnstructured = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+
+	// Prepare the Yaml Reader
+	yamlReader := k8syaml.NewYAMLReader(bufio.NewReader(bytes.NewReader([]byte(fmt.Sprintf("%v", capiYaml)))))
+
+	for err == nil {
```

Comment:
It is good to initialize `err = nil` above this line just to ensure that we don't hit a path where someone allows an escapable error to pass through.